source/main.c

```diff
@@ -17,10 +17,14 @@
 
 #define VERBOSE_LOGGING 0
 
-u8 *dol = NULL;
-int dol_argc = 0;
 #define MAX_NUM_ARGV 1024
-char *dol_argv[MAX_NUM_ARGV];
+
+typedef struct
+{
+    u8 *dol;
+    int dol_argc;
+    char *dol_argv[MAX_NUM_ARGV];
+} BOOT_PAYLOAD;
```

Comment:
Preferably avoid `typedef` (and when you must, suffix the name with `_t`). Type names should be lowercase.